cve_bin_tool/checkers/systemd.py

```diff
@@ -38,20 +38,3 @@ class SystemdChecker(Checker):
     The reason behind this is that these might depend on who packages the file (like it 
     might work on fedora but not on ubuntu)
     """
-
```

Comment:
Let's put a note right in the checker somewhere (maybe here at the bottom) describing the way systemd's version patterns occur, preferably showing the list of strings like was done in #1155.   This is just to help us out in the future in case further refactoring breaks this again.

Equally, we could do something more clever with getting all possible results and sorting them explicitly to make this more robust against changes.  The original cve-bin-tool code actually had something like that in it.

Reply:
Wrote some explanation on what's happening.

Also it seems like the changes in #1227 didn't actually detect the versions in the package from #1155. So I made changes in the first regex pattern. I didn't wan't to bring back what we did in the older fix(reversing the list), since it means we can't use any other utility functions and would have to split, sort and detect versions in the strings in here (due to #1227).

But fortunately the changes I made in the 1st pattern accurately detects the last version string.

Reply:
I have also added that certain package as a condensed download.

Reply:
I don't think we'd actually have to split, you'd just have to use re.findall for systemd and then sort the results.  (You'd do this in an overridden `get_version()` function so we weren't wasting time doing findall on other checkers.)  But I think what you've got right now works and the comment is clear about the multiple version strings, so let's go with the cleaner solution first and save over-riding idea only if they change the sorting.

Reply:
Aah. That could've been more readable than the current implementation. 
Using findall didn't pass through my mind.